Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorSpec.cxx

```diff
@@ -237,7 +244,73 @@ void NoiseCalibratorSpec::sendOutputCcdb(DataAllocator& output)
   auto flName = o2::ccdb::CcdbApi::generateFileName("noise");
   auto image = o2::ccdb::CcdbApi::createObjectImage(&payload, &info);
   info.setFileName(flName);
-  LOG(info) << "Sending object " << info.getPath() << "/" << info.getFileName()
+  LOG(info) << "Sending object CCDB " << info.getPath() << "/" << info.getFileName()
+            << " of size " << image->size()
+            << " bytes, valid for " << info.getStartValidityTimestamp()
+            << " : " << info.getEndValidityTimestamp();
+
+  using clbUtils = o2::calibration::Utils;
+  output.snapshot(Output{clbUtils::gDataOriginCDBPayload, "MFT_NoiseMap", 0}, *image.get());
+  output.snapshot(Output{clbUtils::gDataOriginCDBWrapper, "MFT_NoiseMap", 0}, info);
+}
+
+void NoiseCalibratorSpec::sendOutputCcdbMerge(DataAllocator& output)
+{
+
+  LOG(info) << "CCDB-Merge mode";
+
+  static bool done = false;
+  if (done) {
+    return;
+  }
+  done = true;
+
+  mCalibrator->finalize();
+
+  long tstart = mStart;
+  if (tstart == -1) {
+    tstart = o2::ccdb::getCurrentTimestamp();
+  }
+  long tend = mEnd;
+  if (tend == -1) {
+    constexpr long SECONDSPERYEAR = 365 * 24 * 60 * 60;
+    tend = o2::ccdb::getFutureTimestamp(SECONDSPERYEAR);
+  }
+
+  std::map<std::string, std::string> meta;
+  auto toKeyValPairs = [&meta](std::vector<std::string> const& tokens) {
+    for (auto& token : tokens) {
+      auto keyval = Str::tokenize(token, '=', false);
+      if (keyval.size() != 2) {
+        LOG(error) << "Illegal command-line key/value string: " << token;
+        continue;
+      }
+      Str::trim(keyval[1]);
+      meta[keyval[0]] = keyval[1];
+    }
+  };
+  toKeyValPairs(Str::tokenize(mMeta, ';', true));
+
+  long startTF, endTF;
+
+  auto payload = mCalibrator->getNoiseMap();
+  //  const auto& payload = mCalibrator->getNoiseMap(starTF, endTF); //For TimeSlot calibration
+  map<string, string> headers;
+  map<std::string, std::string> filter;
+  auto* payloadPrev1 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, -1, &headers);
+  long validtime = std::stol(headers["Valid-From"]);
+  auto mergedPL = payload;
+  if (validtime > 0) {
+    validtime = validtime - 60;
```

Comment:
Just to make it clean: if you check `>0`, then you should subtract just 1 (which still will point on prev. object) and not 60.

Reply:
Just to clarify, 
1) Current map is obtained here: auto payload = mCalibrator->getNoiseMap();
2) Previous recorded map is ob tained here : auto* payloadPrev1 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, -1, &headers);
3) The one previous to that is obtained her : auto* payloadPrev2 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, validtime, &headers);

So I am subtracting 60 to go "2 maps in the past"

Reply:
I understand that, what I mean is you should be sure that you can subtract from the SOV of the previous map the one whatever offset you use. If you subtract 60 you should be sure the previous start >60. But it is enough to subtract 1 (then your check >0 is fine). 
This is relevant only if the currently valid map (previous on your definition) happens to be the default one, starting from a small timestamp.